packages/@icon-magic/generate/src/plugins/svg-light-dark.ts

```diff
@@ -0,0 +1,172 @@
+  Icon
+} from '@icon-magic/icon-models';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import Svgo from 'svgo';
+import { DOMImplementation, DOMParser, XMLSerializer } from 'xmldom';
+
+const serializeToString = new XMLSerializer().serializeToString;
+
+export interface SvgLightDarkOptions {
+  lightToken?: string;
+  darkToken?: string;
+}
+
+export const svgLightDark: GeneratePlugin = {
+  name: 'svg-light-dark',
+  fn: async (
+    flavor: Flavor,
+    icon: Icon,
+    params: SvgLightDarkOptions = {},
+  ): Promise<Flavor> => {
+    // Create the output directory
+    const outputPath = icon.getIconOutputPath();
+    const imageSet = flavor.imageset;
+    const colorScheme = flavor.colorScheme;
+
+    /*
+    - If `imageSet` exists and the `colorScheme` equals `dark, run the plugin.
+    - Plugin will skip the `light` flavors because their data can be collected through the `dark` flavor
+    */
+    if (imageSet && colorScheme === 'dark') {
+      // light flavor data
+      const lightFlavor = icon.flavors.get(imageSet);
+
+      // if light flavor doesn't exist, exit out of plugin
+      if (!lightFlavor) {
+        return flavor;
+      } else {
+        const darkFlavorContents = (await flavor.getContents()) as string; // .svg asset's getContents() returns a string (the svg as a string)
+        const lightFlavorContents  = (await lightFlavor.getContents()) as string;
+
+        const lightToken = params.lightToken;
+        const darkToken = params.darkToken;
+
+        const lightSvgo = new Svgo({
+          plugins: [
+            {
+              addAttributesToSVGElement: {
+                attributes: [{display: lightToken || "var(--svg-light-display)"}]
+              }
+            },
+            {
+              removeViewBox: false
+            }
+          ],
+          js2svg: { pretty: true, indent: 2 }
+        });
+
+        const darkSvgo = new Svgo({
+          plugins: [
+            {
+              addAttributesToSVGElement: {
+                attributes: [{display: darkToken || "var(--svg-dark-display)"}]
+              }
+            },
+            {
+              removeViewBox: false
+            }
+          ],
+          js2svg: { pretty: true, indent: 2 }
+        });
+
+        if (lightFlavorContents){
+          // create optimized light/dark svg assets
+          const lightAsset = await lightSvgo.optimize(lightFlavorContents); //returns a string
+          const darkAsset = await darkSvgo.optimize(darkFlavorContents);
+
+          const mixedParentSvg = await domSvgParentGenerate(lightAsset, darkAsset);
+
+          const mixedSvgo = new Svgo({
+            plugins: [
+              {
+                removeViewBox: false
+              }
+            ],
+            js2svg: { pretty: true, indent: 2 }
+          });
+
+          const mixedString = serializeToString(mixedParentSvg);
+          const mixedAsset = await mixedSvgo.optimize(mixedString); //returns a string
```

Comment:
```suggestion
          const mixedAsset = await mixedSvgo.optimize(mixedString); // returns a string
```

Reply:
can you clarify *what* returns a string? svgo.optimize?

Reply:
ah, sorry, I deleted this comment and fixed the one above. optimize does not return a string and mixedString is clear that it is a string..